lib/thread.c

```diff
@@ -164,6 +173,53 @@ static void cpu_record_hash_print(struct hash_bucket *bucket, void *args[])
 		totals->cpu.max = copy.cpu.max;
 }
 
+static void thread_dump_master_statistics(struct vty *vty,
+					  struct thread_master *m)
+{
+	char buf[64];
+
+	if (!thread_collect_stats)
+		return;
+
+	frr_with_mutex (&m->mtx) {
```

Comment:
oh man - I really dislike trying to do io with the mutex held. If this info is useful, we've got to find a way to collect it safely, then format it without holding a lock.

Reply:
@mjstapp My understanding is vty_out() is really not io. vty_out() is just string formatting and writing to a buffer in memory. Please correct me if my understanding is wrong.